tests/e2e/vmlistener_test.go

```diff
@@ -346,6 +352,159 @@ func (s *VMListenerTestParams) TestFailoverAcrossVmOnVsan(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
```

Comment:
I agree we should leverage setup/teardown to reuse the code. However, after developing so many test cases, I think it makes more sense to treat volume creation as part of the test case logic itself. Putting volume creation in setup has caused awkward situation where it's very hard to organize multiple test cases in the same test group. For example, we have created a vmgroupmisc_test.go because it's difficult to add those tests into existing vmgroups_test.go, even though these tests actually belong to the same category.

I definitely agree that we should put common logic in setup/teardown, but if some steps are not really common logic, we shouldn't try to tweak, e.g. to create a new file, just to avoid some code duplication.

For testing code, I feel code clarity should take precedence over code duplication, if we can't achieve both at the same time.